monkey/infection_monkey/network/network_scanner.py

```diff
@@ -118,3 +118,6 @@ def _is_any_ip_in_subnet(ip_addresses, subnet_str):
             if NetworkRange.get_range_obj(subnet_str).is_in_range(ip_address):
                 return True
         return False
+
+    def on_island(self, server):
```

Comment:
Should be either under `monkey/infection_monkey/network/info.py` or `monkey/infection_monkey/utils.py`

Reply:
hmm... I know this method has little to do with network scanning, but the fact that _ip_addresses are private suggests that they should only be accessed from within it's class. That way it stays true to the encapsulation principle. Would be cool to hear what @ShayNehmad or @itaymmguardicore thinks about this.